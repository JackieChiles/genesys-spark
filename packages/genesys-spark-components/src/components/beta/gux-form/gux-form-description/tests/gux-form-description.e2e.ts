import { a11yCheck, newSparkE2EPage } from '../../../../../test/e2eTestUtils';

describe('gux-form-description', () => {
  describe('#render', () => {
    [
      `<gux-form-description><p slot="description">This is a simple paragraph to accompany the form header.</p></gux-form-description>`,
      `<gux-form-description><span slot="description">This is a simple paragraph to accompany the form header.</span></gux-form-description>`
    ].forEach((html, index) => {
      it(`should render the gux-form-description component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-form-description');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
