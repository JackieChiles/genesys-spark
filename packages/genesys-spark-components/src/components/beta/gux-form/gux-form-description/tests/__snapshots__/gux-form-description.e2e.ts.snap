// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-description #render should render the gux-form-description component as expected (1) 1`] = `"<gux-form-description hydrated=""><template shadowrootmode="open"><slot></slot></template><p slot="description">This is a simple paragraph to accompany the form header.</p></gux-form-description>"`;

exports[`gux-form-description #render should render the gux-form-description component as expected (2) 1`] = `"<gux-form-description hydrated=""><template shadowrootmode="open"><slot></slot></template><span slot="description">This is a simple paragraph to accompany the form header.</span></gux-form-description>"`;
